src/test/java/co/com/api/test/ApiGithubRepositoriesTest.java

```diff
@@ -94,28 +104,92 @@ public void apiGetRepositoryFilterRepoTest() {
 
   @Test
   @DisplayName("Check get response with authentication And get repos with filter download")
-  public void apiGetRepositoryFilterDownloadRepoTest() {
+  public void apiGetRepositoryFilterDownloadRepoTest()
+    throws IOException, NoSuchAlgorithmException {
+    final String expectedMd5 = "857c2f0ec3e8c677adfa3bef42170802";
     Response response = given()
       .when()
       .contentType(ContentType.JSON)
       .auth()
       .oauth2(System.getenv("ACCESS_TOKEN"))
       .get("repos");
-    List<GithubDto> values = response
+
+    List<GithubRepoDto> values = response
       .then()
       .extract()
       .body()
       .jsonPath()
-      .getList("findAll { it.name == 'project_api_test' }", GithubDto.class);
+      .getList("findAll { it.name == 'project_api_test' }", GithubRepoDto.class);
 
-    given()
+    byte[] valueArray = given()
       .basePath("")
       .urlEncodingEnabled(false)
       .baseUri(values.get(0).getSvn_url())
       .when()
       .auth()
       .oauth2(System.getenv("ACCESS_TOKEN"))
-      .get("/archive/" + values.get(0).getDefault_Branch() + ".zip");
+      .get("/archive/" + values.get(0).getDefault_branch() + ".zip").asByteArray();
+
+    String fileLocation = "/src/test/resources/downloadFile.zip";
+    Path fileToDeletePath = Paths.get("." + fileLocation);
+    Files.deleteIfExists(fileToDeletePath);
+
+    try {
+      FileOutputStream os = new FileOutputStream(new File("." + fileLocation), false);
+      os.write(valueArray);
+      os.close();
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
```

Comment:
why do you avoid the exception?

Reply:
In cases when the file exists previously the file, and fail the creation, without this try-catch fail the pipeline after some tests

Reply:
why the file should exist?

Reply:
locally , it is possible if file is download previously. in the pipeline y try without the try-cath and the test fail in this line for excepcion